pkg/cloudprovider/aws/instancetype.go

```diff
@@ -60,66 +60,100 @@ func (i *InstanceType) Architecture() string {
 	return fmt.Sprint(aws.StringValueSlice(i.ProcessorInfo.SupportedArchitectures)) // Unrecognized, but used for error printing
 }
 
-func (i *InstanceType) CPU() *resource.Quantity {
-	return resources.Quantity(fmt.Sprint(*i.VCpuInfo.DefaultVCpus))
+func (i *InstanceType) Resources() v1.ResourceList {
+	return v1.ResourceList{
+		v1.ResourceCPU:             i.cpu(),
+		v1.ResourceMemory:          i.memory(),
+		v1.ResourcePods:            i.pods(),
+		v1alpha1.ResourceAWSPodENI: i.awsPodENI(),
+		v1alpha1.ResourceNVIDIAGPU: i.nvidiaGPUs(),
+		v1alpha1.ResourceAMDGPU:    i.amdGPUs(),
+		v1alpha1.ResourceAWSNeuron: i.awsNeurons(),
+	}
+}
+
+func (i *InstanceType) Price() float64 {
+	const (
+		GPUCostWeight      = 5
+		CPUCostWeight      = 1
+		MemoryMBCostWeight = 1
+		MbToGb             = 1 / 1024.0
+	)
+
+	gpuCount := 0.0
+	if i.GpuInfo != nil {
+		for _, gpu := range i.GpuInfo.Gpus {
+			if gpu.Count != nil {
+				gpuCount += float64(*gpu.Count)
+			}
+		}
+	}
+
+	// These are meant to give some relative weighting
```

Comment:
This comment makes me uneasy as a reader. They're meant to do X, but do they? Is it accurate to be more concrete?

Reply:
Maybe I should just remove that comment entirely.  Trying to indicate that it's not a real "price", but people should be able to figure that out easily enough.